mesonbuild/interpreter/interpreter.py

```diff
@@ -614,20 +598,54 @@ def func_import(self, node: mparser.BaseNode, args: T.Tuple[str],
         modname = args[0]
         disabled, required, _ = extract_required_kwarg(kwargs, self.subproject)
         if disabled:
-            return NotFoundExtensionModule()
+            return NotFoundExtensionModule(modname)
+
+        expect_unstable = False
+        # Some tests use "unstable_" instead of "unstable-", and that happens to work because
+        # of implementation details
+        if modname.startswith('unstable'):
+            if modname.startswith('unstable_'):
+                mlog.deprecation(f'Importing unstable modules as "{modname}" instead of "{modname.replace("_", "-", 1)}"',
+                                 location=node)
+            modname = modname[len('unstable') + 1:]  # + 1 to handle the - or _
+            expect_unstable = True
 
-        if modname.startswith('unstable-'):
-            plainname = modname.split('-', 1)[1]
-            try:
-                # check if stable module exists
-                mod = self._import_module(plainname, required)
-                # XXX: this is actually not helpful, since it doesn't do a version check
-                mlog.warning(f'Module {modname} is now stable, please use the {plainname} module instead.')
-                return mod
-            except InvalidArguments:
-                mlog.warning(f'Module {modname} has no backwards or forwards compatibility and might not exist in future releases.', location=node)
-                modname = 'unstable_' + plainname
```

Comment:
After this PR, the warning references "simd" instead of "unstable-simd", which in turn means that a test regex no longer matches.

Reply:
Yup, I added a test and that does happen. I've changed the behavior to allow if required is false (but you get a notfound module), or to error if required and added a test.

Reply:
In the not-found case we still print a warning about its stability?

Reply:
Yup, fixed that too.